src/QldbSession.ts

```diff
@@ -12,136 +12,82 @@
-                if (returnedValue instanceof ResultReadable) {
-                    returnedValue = await Result.bufferResultReadable(returnedValue);
-                }
-                await transaction.commit();
-                return returnedValue;
-            } catch (e) {
-                executionContext.setLastException(e);
-                if (isInvalidSessionException(e)) {
-                    this.closeSession();
-                    throw e;
-                }
-
-                if (!isOccConflictException(e)) {
-                    this._noThrowAbort(transaction);
-                }
-
-                if (executionContext.getExecutionAttempt() >= retryConfig.getRetryLimit()) {
-                    throw e;
-                }
-
-                if (e instanceof StartTransactionError || isRetriableException(e) || isOccConflictException(e)) {
-                    warn(`OCC conflict or retriable exception occurred: ${e}.`);
-                } else {
-                    throw e;
-                }
-            }
-            executionContext.incrementExecutionAttempt();
-            await this._retrySleep(executionContext, retryConfig, transaction);
+    async endSession(): Promise<void> {
+        try {
+            this._isAlive = false;
+            await this._communicator.endSession();
+        } catch (e) {
+            // We will only log issues ending the session, as QLDB will clean them after a timeout.
+            warn(`Errors ending session: ${e}.`);
         }
     }
 
-
-    getSessionToken(): string {
-        return this._communicator.getSessionToken();
-    }
-
-    isSessionOpen(): Boolean {
-        return !this._isClosed;
-    }
-
-    async startTransaction(): Promise<Transaction> {
+    async executeLambda<Type>(
+        transactionLambda: (transactionExecutor: TransactionExecutor) => Promise<Type>
+    ): Promise<Type> {
+        let transaction: Transaction;
+        let transactionId: string = null;
         try {
-            const startTransactionResult: StartTransactionResult = await this._communicator.startTransaction();
-            const transaction: Transaction = new Transaction(
-                this._communicator,
-                startTransactionResult.TransactionId
-            );
-            return transaction;
+            transaction = await this._startTransaction();
+            transactionId = transaction.getTransactionId();
+            const executor: TransactionExecutor = new TransactionExecutor(transaction);
+            const returnedValue: Type = await transactionLambda(executor);
+            await transaction.commit();
+            return returnedValue;
         } catch (e) {
-            if (isBadRequestException(e)) {
-                throw new StartTransactionError(e);
+            const isRetriable: boolean = isRetriableException(e);
+            const isISE: boolean = isInvalidSessionException(e);
+            if (isISE && !isTransactionExpiredException(e)) {
```

Comment:
Again expand the acronym please

Reply:
Done.

Reply:
What @alpian is referring to is the isISE variable, we should expand that to not be the abbrevation.

Reply:
What was missed in the response here is that we cannot expand the name to match the function due to name resolution, and the feeling of keeping this local variable as an acronym would be acceptable.

Reply:
Confused. Why can't `const isISE` be `const isSessionInvalid`?